File: implementation/scipamato/scipamato-core-web/src/test/java/ch/difty/scipamato/core/web/config/CookieTest.java
package ch.difty.scipamato.core.web.config;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.HashSet;
import java.util.Set;

import org.junit.jupiter.api.Test;

public class CookieTest {

    @Test
    public void assertValues() {
        assertThat(Cookie.values()).contains(Cookie.PAPER_LIST_PAGE_MODAL_WINDOW);
    }

    @Test
    public void assertPaperListPageModalWindow() {
        assertThat(Cookie.PAPER_LIST_PAGE_MODAL_WINDOW.getName()).isEqualTo("SciPaMaTo-xmlPasteModal-1");
    }

    @Test
    public void assertAllNamesAreUnique() {
        final Set<String> cookies = new HashSet<>();
        for (final Cookie c : Cookie.values())
            cookies.add(c.getName());
        assertThat(cookies).hasSize(Cookie.values().length);
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/config/CookieTest.kt
package ch.difty.scipamato.core.web.config

import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldContainAll
import org.amshove.kluent.shouldHaveSize
import org.junit.jupiter.api.Test
import java.util.HashSet

internal class CookieTest {

    @Test
    fun assertValues() {
        Cookie.values() shouldContainAll listOf(Cookie.PAPER_LIST_PAGE_MODAL_WINDOW)
    }

    @Test
    fun assertPaperListPageModalWindow() {
        Cookie.PAPER_LIST_PAGE_MODAL_WINDOW.getName() shouldBeEqualTo "SciPaMaTo-xmlPasteModal-1"
    }

    @Test
    fun assertAllNamesAreUnique() {
        val cookies: MutableSet<String> = HashSet()
        for (c in Cookie.values()) cookies.add(c.getName())
        cookies shouldHaveSize Cookie.values().size
    }
}
